manifests/init.pp

```diff
@@ -3,15 +3,15 @@
 # Manage inittab
 #
 class inittab (
-  $default_runlevel                   = 'USE_DEFAULTS',
-  $ensure_ttys1                       = undef,
-  $file_mode                          = '0644',
-  $require_single_user_mode_password  = false,
-  $enable_ctrlaltdel                  = true,
-  $ctrlaltdel_override_path           = 'USE_DEFAULTS',
-  $ctrlaltdel_override_owner          = 'root',
-  $ctrlaltdel_override_group          = 'root',
-  $ctrlaltdel_override_mode           = '0644',
+  Variant[Integer, String] $default_runlevel                 = 'USE_DEFAULTS',
+  Optional[Variant[Boolean,String]] $ensure_ttys1            = undef,
+  Variant[Boolean,String] $file_mode                         = '0644',
+  Variant[Boolean,String] $require_single_user_mode_password = false,
+  Variant[Boolean,String] $enable_ctrlaltdel                 = true,
```

Comment:
We don't need to support stringified booleans, ie `'true'`. Boolean is good.

Reply:
I agree, I did this because of the unit tests and errors they were throwing about such things when I submitted my previous pull requests, due to the tests using random values that didn't fit in with the strict types and the optional values passed to the params being strings initially in some cases. In order to not have to touch the unit tests I relaxed the typing to allow for various values to be passed. I could overhaul the unit tests if you like, but it will require a fair amount more work. This is the case for the file modes also.

Reply:
If you are good with it, that's awesome. If not, I can step in and help with the unit tests. Check out the pam module for a good up to date reference on things.

https://github.com/ghoneycutt/puppet-module-pam/

Reply:
I'm fairly good with puppet-rspec unit tests (although they're black magic to some extent), but I'm out of time to look at them tonight, if you have time to look at the unit tests that would be appreciated, otherwise I'll look at it again later in the week.